code/game/objects/items/explosives/plastique.dm

```diff
@@ -76,22 +76,65 @@
 
 		log_explosion("[key_name(user)] planted [src] at [AREACOORD(user.loc)] with [timer] second fuse.")
 
-		target.overlays += image('icons/obj/items/assemblies.dmi', "plastic-explosive_set_armed")
 		user.visible_message("<span class='warning'>[user] plants [name] on [target]!</span>",
 		"<span class='warning'>You plant [name] on [target]! Timer counting down from [timer].</span>")
-		spawn(timer*10)
-			if(plant_target && !plant_target.gc_destroyed)
-				explosion(location, -1, -1, 3)
-				if(istype(plant_target,/turf/closed/wall))
-					var/turf/closed/wall/W = plant_target
-					W.ChangeTurf(/turf/open/floor/plating)
-				else if(istype(plant_target,/obj/machinery/door))
-					qdel(plant_target)
-				else
-					plant_target.ex_act(1)
-				if(plant_target && !plant_target.gc_destroyed)
-					plant_target.overlays -= image('icons/obj/items/assemblies.dmi', "plastic-explosive_set_armed")
-			qdel(src)
+
+		plant_target = target
+		if(ismovableatom(plant_target))
+			var/atom/movable/T = plant_target
+			T.vis_contents += src
+		detonation_pending = addtimer(CALLBACK(src, .proc/detonate), timer*10, TIMER_STOPPABLE)
+	return
 
 /obj/item/explosive/plastique/attack(mob/M as mob, mob/user as mob, def_zone)
 	return
+
+/obj/item/explosive/plastique/attack_hand(mob/user as mob)
+	if(armed)
+		to_chat(user, "<font color='warning'>Disarm [src] first to remove it!</font>")
+		return
+	return ..()
+
+/obj/item/explosive/plastique/attackby(obj/item/I, mob/user, params)
+	if(ismultitool(I) && armed)
+		if(user.mind?.cm_skills && user.mind.cm_skills.engineer < SKILL_ENGINEER_METAL)
+			user.visible_message("<span class='notice'>[user] fumbles around figuring out how to disarm the [src].</span>",
+			"<span class='notice'>You fumble around figuring out how to disarm the [src].</span>")
+			var/fumbling_time = 30
+			if(!do_after(user, fumbling_time, TRUE, src, BUSY_ICON_UNSKILLED))
+				return
+
+		if(do_after(user, 5 SECONDS, TRUE, plant_target, BUSY_ICON_HOSTILE))
+			if(ismovableatom(plant_target) && plant_target.loc)
+				var/atom/movable/T = plant_target
+				T.vis_contents -= src
+				loc = plant_target.loc
+			else
+				loc = plant_target
+
+			deltimer(detonation_pending)
+			armed = FALSE
+			plant_target = null
+			
+			user.visible_message("<span class='warning'>[user] disarmed [name] that was on [plant_target]!</span>",
+			"<span class='warning'>You disarmed [name] that was on [plant_target]!</span>")
+
+			if(ismob(plant_target))
+				log_combat(user, plant_target, "removed [src] from")
+				log_game("[key_name(usr)] disarmed [src.name] on [key_name(plant_target)].")
+			else
+				log_game("[key_name(user)] disarmed [src.name] on [plant_target.name] at [AREACOORD(plant_target.loc)].")
+	return
+
+/obj/item/explosive/plastique/proc/detonate()
+	if(plant_target && !plant_target.gc_destroyed)
```

Comment:
Early return here. You can go with `if(QDELETED(plant_target))` and then `return`.